### PR TITLE
Coverage: updated source code template

### PR DESCRIPTION
**Pull request by Mikulas**
- fixes line number alignment (this was already implemented but never worked properly)
- fixes copying code should not also copy line numbers
- updated colors, font and spacing

original:
screen shot 2015-08-03 at 12 04 33

proposed:
screen shot 2015-08-03 at 12 03 15

### CODE REVIEW

**Comment by fprochazka** [comment]
Much nicer!

**Comment by JanTvrdik** [comment]
:+1:

**Comment by vojtech-dobes** [comment]
I think it's not good that it looks like diff, because those are completely different things :).

**Comment by JanTvrdik** [comment]
@vojtech-dobes It looked like diff before, now it just looks like a prettier diff :smile:

**Comment by vojtech-dobes** [comment]
That's true... power of habit :).

**Comment by dg** [comment]
This produces invalid HTML soup:

```html
<tr>
	<td class="line-num">1</td>
	<td class="code">
		<code><span style="color: #06B"><span style="color: #000">&lt;?php
	</td>
</tr>
<tr>
	<td class="line-num">2</td>
	<td class="code"></span></td>
</tr>

```

**Comment by Mikulas** [comment]
Good point, I didn't check what markup `highlight_file` produces.

Instead of parsing `highlight_file` output, I've changed the markup to two side-by-side divs; one for line numbers and the other one for the source code.

If this revision is acceptable, let me know and I'll polish the stylesheet a bit.

**Comment by milo** [comment]
:+1: for some facelift

According to https://validator.w3.org it the HTML still invalid.

According to the following screenshot:
screenshot012

- imho, darker background (than white) + border is better for reading, page orientation and saves eyes
- line highlight is shifted according to code

**Comment by Mikulas** [comment]
Fixed: html validity and mismatched line numbering.

I don't agree the original colors are easier to read. Most important lines are the red ones and since syntax highlighting outputs red, the resulting contrast was terrible. I've intentionally used same color schema github uses: we are very accustomed to this look.

**Comment by milo** [comment]
@Mikulas GitHub uses other colors for highlighting. Would you please try the border and some (I don't know, very light gray) background for code blocks?

The reason is, when I'm reading it on a display with a very high resolution, high contrast and brightness (due to sun light in your back) and I toggle 3 source files to show coverage source, the blocks of code just get fused into one white area.

**Comment by milo** [comment]
Btw, how look lines with dead code?

**Comment by Mikulas** [comment]
Updated with slightly darker colors and border.

- original: https://www.dropbox.com/s/204a41iyxqym56w/0807e28b615af7a51acb6903d9f562ccab15fb51.html?dl=0
- proposed@d6dedaa: https://www.dropbox.com/s/dccdutyi5m8wnh3/d6dedaad97f72e914454881aa8863f7bbfdd2fc0.html?dl=0
- proposed@301fe1e: https://www.dropbox.com/s/unyq6ejh6e3x5c4/301fe1ecc7a3649cf848f5422166095d5738f6d0.html?dl=0

**Comment by milo** [comment]
@Mikulas Much better with a border and IMHO even better with `background: #F8F8F8` for the rest. Btw. lines are still shifted.

**Comment by jkuchar** [comment]
Much better! :+1:

File: src/CodeCoverage/Generators/template.phtml [hunk 1]
```diff
@@ -102,46 +148,37 @@
...
-						} elseif (substr($m, -1) !== '/') {
-							$openTags[] = $m;
+		<div class="file" id="fragment<?php echo $id ?>">
+			<div class="lines">
+				<code>
+					<?php
+						$source = highlight_file($info->file, TRUE);
+						$lineCount = count(explode('<br />', $source));
+
+						$prevColor = NULL;
```
Comment (milo) [inline src/CodeCoverage/Generators/template.phtml]: For what it is good now?

**Comment by milo** [comment]
Another style bug.
screenshot013

**Comment by milo** [comment]
@Mikulas Want you to continue on this or may I take over?

**Comment by Mikulas** [comment]
This works for me well enough (well, without rewriting half of code base); feel free to change this however you like.

**Comment by milo** [comment]
Solved by 928cc28e63bb1c10d3e4a78b853758b8c901a7c3

**Pull request closed without merging**